src/main/java/org/dependencytrack/parser/osv/GoogleOSVAdvisoryParser.java

```diff
@@ -0,0 +1,207 @@
+                    final String type = cvss.optString("type", null);
+                    if (type.equalsIgnoreCase("CVSS_V3")) {
+                        advisory.setCvssV3Vector(cvss.optString("score", null));
+                    }
+                    if (type.equalsIgnoreCase("CVSS_V2")) {
+                        advisory.setCvssV2Vector(cvss.optString("score", null));
+                    }
+                }
+            }
+
+            final List<OSVVulnerability> vulnerabilities = parseVulnerabilities(object);
+            advisory.setVulnerabilities(vulnerabilities);
+        }
+        return advisory;
+    }
+
+    private List<OSVVulnerability> parseVulnerabilities(JSONObject object) {
+
+        List<OSVVulnerability> osvVulnerabilityList = new ArrayList<>();
+        final JSONArray vulnerabilities = object.optJSONArray("affected");
+        if (vulnerabilities != null) {
+            for(int i=0; i<vulnerabilities.length(); i++) {
+                osvVulnerabilityList.addAll(parseVulnerabilityRange(vulnerabilities.getJSONObject(i)));
+            }
+        }
+        return osvVulnerabilityList;
+    }
+
+    public List<OSVVulnerability> parseVulnerabilityRange(JSONObject vulnerability) {
+
+        List<OSVVulnerability> osvVulnerabilityList = new ArrayList<>();
+        final JSONArray ranges = vulnerability.optJSONArray("ranges");
+
+        if (ranges != null) {
+            for (int j=0; j<ranges.length(); j++) {
+                final JSONObject range = ranges.getJSONObject(j);
+                osvVulnerabilityList.addAll(parseVersionRanges(vulnerability, range));
+            }
+        } else {
+            osvVulnerabilityList.add(createOSVVulnerability(vulnerability));
+        }
+        return osvVulnerabilityList;
+    }
+
+    private List<OSVVulnerability> parseVersionRanges(JSONObject vulnerability, JSONObject range) {
+
+        final List<OSVVulnerability> osvVulnerabilityList = new ArrayList<>();
+        final JSONArray rangeEvents = range.optJSONArray("events");
+        if(rangeEvents != null) {
+            int k = 0;
+            while (k < rangeEvents.length()) {
+
+                OSVVulnerability osvVulnerability = createOSVVulnerability(vulnerability);
+                JSONObject event = rangeEvents.getJSONObject(k);
+                String lower = event.optString("introduced", null);
+                if(lower != null) {
+                    osvVulnerability.setLowerVersionRange(lower);
+                    k += 1;
+                }
+                if(k < rangeEvents.length()) {
+                    event = rangeEvents.getJSONObject(k);
+                    String upper = event.optString("fixed", null);
+                    if(upper != null) {
+                        osvVulnerability.setUpperVersionRange(upper);
+                        k += 1;
+                    }
+                }
+                osvVulnerabilityList.add(osvVulnerability);
+            }
```

Comment:
This logic is currently causing infinite loops when mirroring the `Linux` or `OSS-Fuzz` ecosystems.

Take https://osv.dev/vulnerability/GSD-2021-1000677 as example:

```json
"ranges": [
    {
        "type": "GIT",
        "repo": "https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/",
        "events": [
            {
                "introduced": "a7d42ddb3099727f58366fa006f850a219cce6c8"
            },
            {
                "limit": "c757c1f1e65d89429db1409429436cf40d47c008"
            }
        ]
    }
],
```

`rangeEvents` is of length `2`. We enter the `while` loop, find the `introduced` property and increment `k` by `1`. It doesn't have a `fixed` property though, so `k` is never incremented to `2` to break the loop.

Another point is that Dependency-Track currently can not deal with commits instead of versions. I wonder if we should skip ranges of type `GIT`?